pkg/apis/workspaces/v1alpha2/buildGuidances.go

```diff
@@ -0,0 +1,65 @@
+package v1alpha2
+
+// BuildGuidanceType describes the type of build guidance.
+// Only one of the following build guidance type may be specified.
+// +kubebuilder:validation:Enum=Dockerfile;SourceToImage
+type BuildGuidanceType string
+
+const (
+	DockerfileBuildGuidanceType    BuildGuidanceType = "Dockerfile"
+	SourceToImageBuildGuidanceType BuildGuidanceType = "SourceToImage"
+)
+
+//+k8s:openapi-gen=true
+type BuildGuidance struct {
+	// Mandatory name that allows referencing the buid guidance from other elements or from
+	// an external devfile that may reference this build guidance through a parent or a plugin.
+	Name               string `json:"name"`
+	BuildGuidanceUnion `json:",inline"`
+}
+
+// +union
+type BuildGuidanceUnion struct {
+	// Type of build guidance
+	//
+	// +unionDiscriminator
+	// +optional
+	BuildGuidanceType `json:"buildGuidanceType,omitempty"`
+
+	// +optional
+	// Allows specifying a dockerfile to initiate build
+	Dockerfile *Dockerfile `json:"dockerfile,omitempty"`
+
+	// +optional
+	// Allows specifying a builder image to initiate s2i (SourceToImage) build
+	SourceToImage *SourceToImage `json:"s2i,omitempty"`
+}
+
+type Dockerfile struct {
+	// Dockerfile location which can be an URL or a path relative to buildContext
+	DockerfileLocation string `json:"dockerfileLocation"`
+
+	// +optional
+	// Path of source directory to establish build context.  Default to the top level directory.
+	BuildContext string `json:"buildContext,omitempty"`
```

Comment:
I am not sure if we want to continue to work on this branch for Build Guidance or open a new PR but for record, I had a question:

What is the difference between this `BuildContext` and the one provided from the Console POC import from devfile form https://github.com/openshift/console/pull/6321

![Screen Shot 2020-11-16 at 12 29 19 PM](https://user-images.githubusercontent.com/31771087/99319691-fbef2c00-2837-11eb-8d33-8258e73cd032.png)